### ISSUE TITLE
[camera] Official example throws errors when used in any project other than cloned plugin repository

### ISSUE THREAD

**Issue by TahaTesser** [flutter/flutter#73844]
### Steps to Reproduce
1. Create a project on any Flutter channel `flutter create demoapp`
2. Adding the following packages to `pubspec.yaml`
```yaml
  camera: ^0.6.4+5
  video_player: ^1.0.1
```
4. Adding the following permissions to `demoapp/android/app/src/main/AndroidManifest.xml` to the test example on the Android platform or iOS permissions (doesn't matter, it's a  code sample issue). 
```xml
    <uses-feature
      android:name="android.hardware.camera"
      android:required="true"/>

    <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.FLASHLIGHT"/> 
```
5. Now copy the [official camera example](https://github.com/flutter/plugins/blob/master/packages/camera/camera/example/lib/main.dart) and paste it in main.dart
6. Execute `flutter pub get`
7. Run the app on Android.

### Expected Results
`main.dart` doesn't show any errors or red lines and the app runs on Android

### Actual Results
`main.dart` throws a bunch of errors and can also be verified by running `flutter analyze`, the app doesn't run. 
```bash
tahatesser@Tahas-MacBook-Pro camera_example % flutter analyze
Analyzing camera_example...                                             

  error • The getter 'focusMode' isn't defined for the type 'CameraValue' • lib/main.dart:409:35 • undefined_getter
  error • Undefined name 'FocusMode' • lib/main.dart:409:48 • undefined_identifier
  error • The getter 'focusMode' isn't defined for the type 'CameraValue' • lib/main.dart:414:35 • undefined_getter
  error • Undefined name 'FocusMode' • lib/main.dart:414:48 • undefined_identifier
  error • Undefined name 'FocusMode' • lib/main.dart:437:61 • undefined_identifier
  error • The method 'setFocusPoint' isn't defined for the type 'CameraController' • lib/main.dart:440:58 •
         undefined_method
  error • Undefined name 'FocusMode' • lib/main.dart:448:61 • undefined_identifier
  error • The method 'setFocusPoint' isn't defined for the type 'CameraController' • lib/main.dart:550:16 •
         undefined_method
  error • The named parameter 'imageFormatGroup' isn't defined • lib/main.dart:561:7 • undefined_named_parameter
  error • There's no constant named 'jpeg' in 'ImageFormatGroup' • lib/main.dart:561:42 • undefined_enum_constant
  error • Undefined class 'FocusMode' • lib/main.dart:651:36 • undefined_class
  error • Undefined class 'FocusMode' • lib/main.dart:777:29 • undefined_class
  error • The method 'setFocusMode' isn't defined for the type 'CameraController' • lib/main.dart:779:24 •
         undefined_method
   info • Unused import: 'package:triage/main.dart' • test/widget_test.dart:11:8 • unused_import

14 issues found. (ran in 2.3s)
```

And overriding [camera_platform_interface](https://pub.dev/packages/camera_platform_interface) doesn't help.
```yaml
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^0.1.3
  camera: ^0.6.4+5
  video_player: ^1.0.1

dependency_overrides:
  camera_platform_interface: ">=1.0.4 <1.5.0" 
```

Additional notes:
1. This blocks me from using an official example in any Flutter app and only having to resort to the cloned repo for using this example.

### Steps to Reproduce
1. Clone `plugins` repo, `git clone https://github.com/flutter/plugins.git`.
2. Open `plugins/packages/camera/camera/example/` in your iDE and run the app on Android or IOS.

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 1.22.5, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.22.5 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • Android SDK built for x86 (mobile) • emulator-5554             • android-x86 • Android 10 (API 29) (emulator)
    • Taha’s iPad (mobile)               • 00008020-000255113EE8402E • ios         • iOS 14.3

! Doctor found issues in 1 category.
```
```bash
[✓] Flutter (Channel beta, 1.25.0-8.2.pre, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.25.0-8.2.pre at /Users/tahatesser/Code/flutter_beta
    • Framework revision b0a2299859 (8 days ago), 2021-01-05 12:34:13 -0800
    • Engine revision 92ae191c17
    • Dart version 2.12.0 (build 2.12.0-133.2.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (3 available)
    • Android SDK built for x86 (mobile) • emulator-5554             • android-x86    • Android 10 (API 29) (emulator)
    • Taha’s iPad (mobile)               • 00008020-000255113EE8402E • ios            • iOS 14.3
    • Chrome (web)                       • chrome                    • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```
```bash
[✓] Flutter (Channel dev, 1.26.0-1.0.pre, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-1.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision 63062a6443 (4 weeks ago), 2020-12-13 23:19:13 +0800
    • Engine revision 4797b06652
    • Dart version 2.12.0 (build 2.12.0-141.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554             • android-x86    • Android 10 (API 29) (emulator)
    • Taha’s iPad (mobile)               • 00008020-000255113EE8402E • ios            • iOS 14.3
    • macOS (desktop)                    • macos                     • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)                       • chrome                    • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```
```bash
[✓] Flutter (Channel master, 1.26.0-2.0.pre.313, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-2.0.pre.313 at /Users/tahatesser/Code/flutter_master
    • Framework revision 1cb0a24a46 (5 hours ago), 2021-01-12 21:39:27 -0800
    • Engine revision 10cee6abcb
    • Dart version 2.12.0 (build 2.12.0-214.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554             • android-x86    • Android 10 (API 29) (emulator)
    • Taha’s iPad (mobile)               • 00008020-000255113EE8402E • ios            • iOS 14.3
    • macOS (desktop)                    • macos                     • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)                       • chrome                    • web-javascript • Google Chrome 87.0.4280.141

• No issues found!
```

</details>

Thank you

**Comment by mvanbeusekom** [comment]
@TahaTesser this is probably because you are using the code from the `master` branch which contains changes that have not yet been released to pub.dev (version of the `camera/camera/pubspec.yaml` on `master` is 0.6.6). 

If you would use the example code that belongs to the 0.6.4+5 version from GitHub (select tag `camera-v0.6.4+5` or check this [link](https://github.com/flutter/plugins/blob/camera-v0.6.4%2B5/packages/camera/camera/example/lib/main.dart)) all should be fine.

**Comment by TahaTesser** [comment]
Hi @mvanbeusekom 
Thanks! That helps a lot, I've seen similar issues with other packages so that's due to the unreleased  `master` branch. 
I would expect `maste  `to be up to date with `pub.dev`, and in the past, my PR was merged into `master` and was also live pub.dev in a short time hence I tried overriding interface too. 

Closing as resolved and assumed this applies to `url_lauincher` and others 
Thank you
